src/vanilla.cpp

```diff
@@ -67,19 +67,18 @@ Rcpp::List europeanOptionEngine(std::string type,
     boost::shared_ptr<QuantLib::StrikedTypePayoff> payoff(new QuantLib::PlainVanillaPayoff(optionType, strike));
     
     if (withDividends) {
-        Rcpp::NumericVector divvalues(discreteDividends), divtimes(discreteDividendsTimeUntil);
-        int n = divvalues.size();
+        std::vector<double> discDividends = Rcpp::as<std::vector<double> >(discreteDividends);
+        std::vector<double> divtimes = Rcpp::as<std::vector<double> >(discreteDividendsTimeUntil);
```

Comment:
This instantiation is entirely equivalent, but possibly worse because of a potential copy.   

Your variable naming is less consistent as you mix lower and uppercase (not that that in itself requires a commit let a lone a PR).